fastfield_codecs/src/column.rs

```diff
@@ -137,37 +137,53 @@ where V: AsRef<[T]> + ?Sized
     }
 }
 
-struct MonotonicMappingColumn<C, T, Input> {
+struct MonotonicMappingColumn<C, T, U, Input> {
     from_column: C,
     monotonic_mapping: T,
+    monotonic_mapping_inv: U,
     _phantom: PhantomData<Input>,
 }
 
 /// Creates a view of a column transformed by a monotonic mapping.
-pub fn monotonic_map_column<C, T, Input: PartialOrd, Output: PartialOrd>(
+/// E.g. apply a gcd monotonic_mapping([100, 200, 300]) == [1, 2, 3]
+/// The provided mappings need to be the inverse of each other.
+///
+/// The inverse of the mapping is required for:
+/// `fn get_between_vals(&self, range: RangeInclusive<T>) -> Vec<u64> `
+/// The user provides the original value range and we need to monotonic map them in the same way the
+/// serialization does before calling the underlying column.
+///
+/// Note that when opening a codec, the monotonic_mapping should be the inverse of the mapping
+/// during serialization. And therefore the monotonic_mapping_inv when opening is the same as
+/// monotonic_mapping during serialization.
+pub fn monotonic_map_column<C, T, U, Input: PartialOrd, Output: PartialOrd + Clone>(
     from_column: C,
     monotonic_mapping: T,
+    monotonic_mapping_inv: U,
```

Comment:
I think this is a great idea to introduce the notion of inverse mapping.

Can we go a step further and introduce a `StrictlyMonotonicMapping` trait.

```rust
pub trait StrictlyMonotonicFn<Input, Ouput> {
    fn mapping(&self, inp: Input) -> Output;
    fn inverse(&self, inp: Input) -> Output;
}
```

Two benefits:
- The code will be a tad clearer.
- The contract will be clearer.
- The GCD stuff will be implemented once and for all and reuse instead of being implemented twice with a useless unimplemented!().

Reply:
Yes, I also thought about that, it's probably clearer.

Input and Output are flipped for serialization and opening, so I think we still have two implementations. Maybe we can have an inverse over the trait for that ... but I'm not sure that's very helpful for clarity

Reply:
Okay done, it's a little more verbose than before. I chose `<External, Internal>`, since it's less ambigious. Since we have nameable types now we could evaluate to move it to `Column` and drop the monotonic column wrapping, which does cost some performance.